Mathlib/CategoryTheory/Shift/Pullback.lean

```diff
@@ -103,4 +118,47 @@ lemma pullbackShiftFunctorAdd'_hom_app :
     ← Functor.map_comp, Iso.hom_inv_id_app, Functor.map_id]
   rfl
 
+namespace Functor
+
+variable {D : Type*} [Category D] [HasShift D B] (F : C ⥤ D) [F.CommShift B]
+
+/-- If `F : C ⥤ D` commutes with the shifts on `C` and `D`, then it also commutes with
+their pullbacks by an additive map.
+-/
+@[simps]
+noncomputable def commShiftPullback :
+    F.CommShift A (C := PullbackShift C φ) (D := PullbackShift D φ) where
+  iso a := isoWhiskerRight (pullbackShiftIso C φ a (φ a) rfl) F ≪≫
+    CommShift.iso (F := F) (φ a) ≪≫ isoWhiskerLeft _  (pullbackShiftIso D φ a (φ a) rfl).symm
+  zero := by
+    ext
+    simp only [comp_obj, Iso.trans_hom, isoWhiskerRight_hom, isoWhiskerLeft_hom, Iso.symm_hom,
+      NatTrans.comp_app, whiskerRight_app, whiskerLeft_app, CommShift.isoZero_hom_app,
+      pullbackShiftFunctorZero'_hom_app, id_obj, map_comp, pullbackShiftFunctorZero'_inv_app, assoc]
+    conv_lhs => congr; rfl; congr; change (F.commShiftIso (φ 0)).hom.app _
+                rw [F.commShiftIso_zero' (A := B) (φ 0) (by rw [map_zero])]
+    simp only [CommShift.isoZero'_hom_app, assoc]
```

Comment:
I would remove the `simps` attribute, because we should introduce an equational lemma where `φ a` is replace by any `b` such that `b = φ a`:
```suggestion
noncomputable def commShiftPullback :
    F.CommShift A (C := PullbackShift C φ) (D := PullbackShift D φ) where
  iso a := isoWhiskerRight (pullbackShiftIso C φ a (φ a) rfl) F ≪≫
    F.commShiftIso (φ a) ≪≫ isoWhiskerLeft _  (pullbackShiftIso D φ a (φ a) rfl).symm
  zero := by
    ext
    dsimp
    simp only [F.commShiftIso_zero' (A := B) (φ 0) (by rw [map_zero]), CommShift.isoZero'_hom_app,
      assoc, CommShift.isoZero_hom_app, pullbackShiftFunctorZero'_hom_app, map_comp,
      pullbackShiftFunctorZero'_inv_app]
    dsimp
    rfl
```

Reply:
Yeah ok.